syntax = "proto3";

//使用时注意，命名空间经常忘
package us;

option cc_generic_services = true;

message ReturnCode
{
    int32 errcode =1 ;  //错误码
    bytes errdesc= 2;   //错误描述
}

//Login方法请求参数
message LoginRequest
{
    bytes name =1;
    bytes pwd=2;
}   

//Login方法返回值
message LoginResponse
{
    ReturnCode result =1;
    bool sucess = 2;
}
//---
//添加一个Register注册方法
message RegisterRequest
{
    uint32 id  = 1;
    bytes name = 2;
    bytes pwd = 3 ;
}

message RegisterResponse
{
    ReturnCode result =1;
    bool sucess = 2;
}

//标识Login方法
service UserServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
}